### ISSUE TITLE
az image builder create claims invalid JSON

### ISSUE THREAD

**Issue by bweissman** [Azure/azure-cli#18729]
**Describe the bug**
when running az image builder create, when passing a template JSON, the CLI always claims that the JSON is invalid.

**To Reproduce**
az image builder create -g $imageResourceGroup -n $TemplateName --image-template ImageTemplate.json

Same for any JSON - you can for example use this one: https://raw.githubusercontent.com/azure/azvmimagebuilder/master/quickquickstarts/0_Creating_a_Custom_Windows_Managed_Image/helloImageTemplateWin.json

(setting subscription etc. to valid values)

I tried with 10 different files - all same result and all are valid JSON.

**Expected behavior**
This should lead to the same resource creation as this:

az resource create --resource-group $imageResourceGroup --properties "@./ImageTemplate.json" --is-full-object `
                   --resource-type Microsoft.VirtualMachineImages/imageTemplates -n $TemplateName

the az resource command works fine - passing in identical JSON.

**Environment summary**
az cli 2.25.0 on Windows 10

**Comment by yonzhan** [comment]
Compute

**Comment by bweissman** [comment]
The issue is coming from encoding. This command requires ascii encoding whereas az resource create takes any.

**Comment by zhoxing-ms** [comment]
@bweissman Yes, I would like to ask what is the specific error message? Is it accurate?

**Comment by bweissman** [comment]
The error message says it's invalid JSON. One might argue if encoding really qualifies as invalid.

Ideally, this command would also accept the same encodings as "az resource create".

Otherwise, it should go to the docs which encoding(s) are supported.

**Comment by tylerjharden** [comment]
Any update on this?
